### DISCUSSION TITLE
Preserving file names after uploading it through Ansible forms

### DISCUSSION THREAD

**Discussion by nx2804** [ansibleguy76/ansibleforms#134] [Q&A] [answered]
Hi Team,

Using file field type when i upload a file the name is getting modified. instead i wanted the file name to be preserved

also i wanted to generate a drop down which list all the files in certain directory. Will it be possible to get that using Ansible forms?

Thanks,

**Comment by nx2804** [comment]
or any expressions would just do a rename with executing playbook

**Comment by vcmirko** [comment, marked as answer]
> Hi Team,
> 
> Using file field type when i upload a file the name is getting modified. instead i wanted the file name to be preserved
> 
> also i wanted to generate a drop down which list all the files in certain directory. Will it be possible to get that using Ansible forms?
> 
> Thanks,

That's correct, the file is uploaded under a temp filename, but extravars are sent with the original name so that you can move/copy/rename the file in ansible.  So you can rename it back to the original name.

**Comment by vcmirko** [comment]
> Hi Team,
> 
> Using file field type when i upload a file the name is getting modified. instead i wanted the file name to be preserved
> 
> also i wanted to generate a drop down which list all the files in certain directory. Will it be possible to get that using Ansible forms?
> 
> Thanks,

Regarding the dropdown, there is a remote expression fn.fnSsh that allows you to ssh to a machine, run a command, and parse the output, you can do an "ls" to get the list of files.

**Comment by nx2804** [comment]
But I want this to be done with out using playbook or manual

Any expressions support this for renaming the file immediately after upload

**Reply by ansibleguy76** [reply]
I understand, but I have to make temp filenames to avoid duplicate collisions.

**Comment by nx2804** [comment]
Where can I find the list of functions available

**Comment by ansibleguy76** [comment]
https://ansibleforms.com/expressions/#remote-expressions

also consider using discussions for questions so we can keep issues for actual issues.
